drivers/bluetooth/hci/hci_stm32wba.c

```diff
@@ -380,7 +394,91 @@ static int bt_hci_stm32wba_open(const struct device *dev, bt_hci_recv_t recv)
 	return ret;
 }
 
+#if defined(CONFIG_BT_HCI_SETUP)
+
+bt_addr_t *bt_get_ble_addr(void)
+{
+	bt_addr_t *bd_addr;
+	uint32_t udn;
+	uint32_t company_id;
+	uint32_t device_id;
+
+	/* Get the 64 bit Unique Device Number UID */
+	/* The UID is used by firmware to derive */
+	/* 48-bit Device Address EUI-48 */
+	udn = LL_FLASH_GetUDN();
+
+	if (udn != 0xFFFFFFFF) {
+		/* Get the ST Company ID */
+		company_id = LL_FLASH_GetSTCompanyID();
+		/* Get the STM32 Device ID */
+		device_id = LL_FLASH_GetDeviceID();
+
+		/*
+		 * Public Address with the ST company ID
+		 * bit[47:24] : 24bits (OUI) equal to the company ID
+		 * bit[23:16] : Device ID.
+		 * bit[15:0] : The last 16bits from the UDN
+		 * Note: In order to use the Public Address in a final product, a dedicated
+		 * 24bits company ID (OUI) shall be bought.
+		 */
+
+		bd_addr_udn.val[0] = (uint8_t)(udn & 0x000000FF);
+		bd_addr_udn.val[1] = (uint8_t)((udn & 0x0000FF00) >> 8);
+		bd_addr_udn.val[2] = (uint8_t)device_id;
+		bd_addr_udn.val[3] = (uint8_t)(company_id & 0x000000FF);
+		bd_addr_udn.val[4] = (uint8_t)((company_id & 0x0000FF00) >> 8);
+		bd_addr_udn.val[5] = (uint8_t)((company_id & 0x00FF0000) >> 16);
+		bd_addr = &bd_addr_udn;
+	} else {
+		bd_addr = &bd_addr_dflt;
+	}
+
+	return bd_addr;
+}
+
+static int bt_hci_stm32wba_setup(const struct device *dev,
+				 const struct bt_hci_setup_params *params)
+{
+	bt_addr_t *uid_addr;
+	struct aci_set_ble_addr *param;
+	struct net_buf *buf, *rsp;
+	int err;
+
+	uid_addr = bt_get_ble_addr();
+	if (!uid_addr) {
+		return -ENOMSG;
+	}
+
+	buf = bt_hci_cmd_create(ACI_HAL_WRITE_CONFIG_DATA, sizeof(*param));
+
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	param = net_buf_add(buf, sizeof(*param));
+	param->config_offset = HCI_CONFIG_DATA_PUBADDR_OFFSET;
+	param->length = 6;
+
+	if (bt_addr_eq(&(params->public_addr), BT_ADDR_ANY)) {
+		bt_addr_copy((bt_addr_t *)param->value, uid_addr);
+	} else {
+		bt_addr_copy((bt_addr_t *)param->value, &(params->public_addr));
+	}
+
+	err = bt_hci_cmd_send_sync(ACI_HAL_WRITE_CONFIG_DATA, buf, &rsp);
+	if (err) {
+		return err;
+	}
+	net_buf_unref(rsp);
```

Comment:
Since you're not doing anything with `rsp` you can just pass NULL to the function. Also a minor style thing: please keep an empty line after the final `}` of the if-statement (i.e. before `return 0;` in this case).

Reply:
ok,  totally agree